Fix compiling with gcc-2.95.x:

-----
newgrp.c: In function `main':
newgrp.c:459: parse error before `child'
newgrp.c:467: `child' undeclared (first use in this function)
newgrp.c:467: (Each undeclared identifier is reported only once
newgrp.c:467: for each function it appears in.)
newgrp.c:476: `pid' undeclared (first use in this function)
make[2]: *** [newgrp.o] Error 1
-----

--- shadow-4.0.12/src/newgrp.c	2005-08-24 13:30:51.000000000 +0200
+++ shadow-4.0.12.az/src/newgrp.c	2005-08-24 13:31:01.000000000 +0200
@@ -424,6 +424,7 @@
 	if (getdef_bool ("SYSLOG_SG_ENAB")) {
 		char *loginname = getlogin ();
 		char *tty = ttyname (0);
+		pid_t child, pid;
 
 		if (loginname != NULL)
 			loginname = xstrdup (loginname);
@@ -456,7 +457,6 @@
 		 * avoid any possibility of the parent being stopped when it
 		 * receives SIGCHLD from the terminating subshell.  -- JWP
 		 */
-		pid_t child, pid;
 
 		signal (SIGINT, SIG_IGN);
 		signal (SIGQUIT, SIG_IGN);
